### PR TITLE
dotdigital section updates for 2.4.2 release

### PR DESCRIPTION
**Pull request by sta1r**
## Purpose of this pull request

This PR updates a number of areas in the dotdigital merch docs, including:
- a new section for Transactional SMS (note this will be dependent on our new `Dotdigitalgroup_Sms` module being installed)
- new sections for Image Types and System Alerts
- an updated Transactional Emails section that now contains the old Email Templates section 

## Magento release version

- [x] 2.4.2

## Product editions

Is this update specific to a product edition?

- [ ] Magento Open Source only
- [ ] Magento Commerce only

Is this update specific to an installed feature extension

- [ ] B2B extension
- [x] Other feature set, please specify:
-- `Dotdigitalgroup_Sms` module must be installed to configure and use the Transactional SMS page

## Affected documentation pages

- https://docs.magento.com/user-guide/configuration/dotdigital/accounts.html
- https://docs.magento.com/user-guide/configuration/dotdigital/chat-settings.html
- https://docs.magento.com/user-guide/configuration/dotdigital/abandoned-carts.html
- https://docs.magento.com/user-guide/configuration/dotdigital/transactional-emails.html
- https://docs.magento.com/user-guide/configuration/dotdigital/configuration.html
- https://docs.magento.com/user-guide/configuration/dotdigital/developer.html

whatsnew
Updates to for the latest release of the dotdigital bundled extension:
- A new page for [Transactional SMS](https://docs.magento.com/user-guide/configuration/dotdigital/transactional-sms.html) (dependent on the new `Dotdigitalgroup_Sms` module being installed)
- New sections for [Image Types](https://docs.magento.com/user-guide/configuration/dotdigital/configuration.html#image-types) and [System Alerts](https://docs.magento.com/user-guide/configuration/dotdigital/developer.html#system-alerts)
- A new [Transactional Emails](https://docs.magento.com/user-guide/configuration/dotdigital/transactional-emails.html) section that replaces the old _Email Templates_ section

### CODE REVIEW

File: src/configuration/dotdigital/transactional-emails.md [hunk 1]
```diff
@@ -4,19 +4,64 @@ redirect_from:
...
+|New Account Confirmation Key|Store View|The campaign that sends a confirmation key when customers open a new account. Options: (All available campaigns)|
+|New Account Confirmation|Store View|The campaign that sends a confirmation email to customers who open a new account. Options: (All available campaigns)|
+|Forgot Password|Store View|The campaign that sends an email to customers who forget their passwords. Options: (All available campaigns)|
+|Remind Password|Store View|The campaign that sends a password reminder email to customers who have forgotten their passwords. Options: (All available campaigns)|
+|Wish List Sharing|Store View|The campaign that sends a confirmation email when the customer shares a wish list. Options: (All available campaigns)|
+|Forgot Admin Password|Store View|The campaign that sends a password reminder email to Admin users who have forgotten their passwords. Options: (All available campaigns)|
+|**Newsletter**|||
+|Success Email Template|Website|The campaign that sends a confirmation email at the completion of a subscription request.Options: (All available campaigns)|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-emails.md:40]:
```suggestion
|Success Email Template|Website|The campaign that sends a confirmation email at the completion of a subscription request. Options: (All available campaigns)|
```

File: src/configuration/dotdigital/transactional-emails.md [hunk 2]
```diff
@@ -4,19 +4,64 @@ redirect_from:
...
+|Invoice Update|Website|The campaign that notifies customers when an invoice is updated.  Options: (All available campaigns)|
+|Invoice Update Guest|Website|The campaign that notifies guests when an invoice is updated. Options: (All available campaigns)|
+|New Credit Credit Memo|Website|The campaign that notifies customers when a credit memo is generated.  Options: (All available campaigns)|
+|New Credit Credit Memo Guest|Website|The campaign that notifies guests when a credit memo is generated. Options: (All available campaigns)|
+|Credit Memo Update|Website|The campaign that notifies customers when a credit memo is updated. Options: (All available campaigns)|
+|Credit Memo Update Guest|Website|The campaign that notifies guests when a credit memo is updated. Options: (All available campaigns)|
+|**Contact**|||
+|New Contact Form|Website|The campaign that sends confirmation email when a new contact record is submitted. Options: (All available campaigns)|
+|**Email**|||
+|Send Product to Friend|Website|The campaign that sends notification when a customer shares a product to a friend. Options: (All available campaigns)|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-emails.md:62]:
```suggestion
|Send Product to Friend|Website|The campaign that sends notification when a customer shares a product with a friend. Options: (All available campaigns)|
```

File: src/configuration/dotdigital/transactional-emails.md [hunk 3]
```diff
@@ -4,19 +4,64 @@ redirect_from:
...
+|**Product Alerts**|||
+|Stock Alert|Website|The campaign that sends notification to a customer of a stock alert. Options: (All available campaigns)|
+|Product Price Alert|Website|The campaign that sends notification to a customer of a product price alert. Options: (All available campaigns)|
+|**Additional Templates**|||
+|Sync additional campaigns as templates|Website|Maps additional campaigns as templates. Synced templates can be selected from Magento.<br /><span class="btn">Sync New Campaign</span>Map a new campaign to appear in lists of available campaign templates. Options: <br />**Campaign**: Lists mapped campaigns by name.<br />**Action**: Lists the action that triggers the campaign.|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-emails.md:67]:
```suggestion
|Sync additional campaigns as templates|Website|Maps additional campaigns as templates. Synced templates can be selected from Magento.<br /><span class="btn">Sync New Campaign</span> Map a new campaign to appear in lists of available campaign templates. Options: <br />**Campaign**: Lists mapped campaigns by name.<br />**Action**: Lists the action that triggers the campaign.|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## Transactional SMS Settings
+
+![]({% link images/images/config-engagement-cloud-transactional-sms.png %}){: .zoom}
+_Transactional SMS Settings_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enabled|Store View|Specifies if the dotdigital Transactional SMS functionality is enabled for your Magento store. Options: Yes / No|
+|Enable Phone Number Validation|Store View|Specifies if phone number validation is turned on at checkout and on the account dashboard. This ensures any phone number that’s entered by customers is in a valid format and has a correct country code. Options: Yes / No|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:15]:
```suggestion
|Enable Phone Number Validation|Store View|Specifies if phone number validation is turned on at checkout and on the account dashboard. Ensures that all phone numbers entered by customers have a valid format and country code. Options: Yes / No|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+## SMS Templates
+
+![]({% link images/images/config-engagement-cloud-sms-templates.png %}){: .zoom}
+_SMS Templates_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|**New Order Confirmation**|||
+|Enabled|Store View|Specifies if the new order confirmation SMS is sent to your customers when new orders are created. Options: Yes / No|
+|Message|Store View|Defines what message template is used to send the new order confirmation SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}. Options: Yes / No|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:27]:
```suggestion
|Message|Store View|Defines the message template that is used to send the new order confirmation SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}. Options: Yes / No|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+|**Order Update**|||
+|Enabled|Store View|Specifies if the order update SMS is sent to your customers when an order is updated (on hold or cancelled). Options: Yes / No|
+|Message|Store View|Defines what message template is used to send the order update SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}.|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:30]:
```suggestion
|Message|Store View|Defines the message template that is used to send the order update SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}.|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+|**New Shipment**|||
+|Enabled|Store View|Specifies if the new shipment SMS is sent to your customers when a new shipment is created. Options: Yes / No|
+|Message|Store View|Defines what message template is used to send the new shipment SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}, {{ tracking_number }}, {{ tracking_carrier }}.|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:33]:
```suggestion
|Message|Store View|Defines the message template that is used to send the new shipment SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}, {{ tracking_number }}, {{ tracking_carrier }}.|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+|**New Shipment Update**|||
+|Enabled|Store View|Specifies if the new shipment update SMS is sent to your customers when a shipment is being updated. Options: Yes / No|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:35]:
```suggestion
|Enabled|Store View|Specifies if the new shipment update SMS is sent to your customers when a shipment is updated. Options: Yes / No|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+|Message|Store View|Defines what message template is used to send the new shipment update SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}, {{ tracking_number }}, {{ tracking_carrier }}.|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:36]:
```suggestion
|Message|Store View|Defines the message template that is used to send the new shipment update SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}, {{ tracking_number }}, {{ tracking_carrier }}.|
```

File: src/configuration/dotdigital/transactional-sms.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+|**New Credit Memo**|||
+|Enabled|Store View|Specifies if the new credit memo SMS is sent to your customers when a new refund is created. Options: Yes / No|
+|Message|Store View|Defines what message template is used to send the new credit memo SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}, {{ refund_amount }}.|
```
Comment (ccondra) [inline src/configuration/dotdigital/transactional-sms.md:39]:
```suggestion
|Message|Store View|Defines the message template that is used to send the new credit memo SMS. Supported variables are: {{ first_name }}, {{ last_name }}, {{ email }}, {{ store_name }}, {{ order_id }}, {{ order_status }}, {{ refund_amount }}.|
```

File: src/configuration/dotdigital/abandoned-carts.md [hunk 11]
```diff
@@ -13,7 +13,7 @@ _Abandoned Cart Program_
 
 |Field|[Scope]({% link configuration/scope.md %})|Description|
 |--- |--- |--- |
-|Enrol abandoned cart to|Store View|Determines the [CartInsight](https://support.dotdigital.com/hc/en-gb/articles/360000456890-Getting-started) program that is triggered when a cart is abandoned. This option is disabled until an active abandoned cart program is available in your dotdigital  account. Options: (Available programs)|
+|Enrol abandoned cart to|Store View|Determines the [automation program](https://support.dotdigital.com/hc/en-gb/articles/360000456890-Getting-started) that is triggered when a cart is abandoned. This option is disabled until an active abandoned cart program is available in your dotdigital  account. Options: (Available programs)|
```
Comment (ccondra) [inline src/configuration/dotdigital/abandoned-carts.md:16]:
```suggestion
|Enrol abandoned cart to|Store View|Determines the [automation program](https://support.dotdigital.com/hc/en-gb/articles/360000456890-Getting-started) that is triggered when a cart is abandoned. This option is disabled until an active abandoned cart program is available in your dotdigital account. Options: (Available programs)|
```

File: src/configuration/dotdigital/chat-settings.md [hunk 12]
```diff
@@ -16,6 +16,6 @@ _[Chat Settings - Enabled]({% link marketing/dotdigital/setup.md %})_
 
 |Field|[Scope]({% link configuration/scope.md %})|Description|
 |--- |--- |--- |
-|Enabled|Website|Determines if Chat is enabled in your dotdigital account. When enabled, the Chat Widget and Chat Teams configuration buttons become available. Options: Yes / No|
-|||Configure Chat Widget<br /><span class="btn">Configure</span> Opens the Widget tab from the Account page of your dotdigital account.|
-|||Configure Chat Teams<br /><span class="btn">Configure</span> Opens the Teams tab from the Account page of your dotdigital account.|
+|Enabled|Website|Determines if Chat is enabled on your Magento website. When enabled, the Chat widget appears on the pages of your website and the configuration buttons become available. Options: Yes / No|
```
Comment (ccondra) [inline src/configuration/dotdigital/chat-settings.md:19]:
```suggestion
|Enabled|Website|Determines if Chat is enabled for your Magento website. When enabled, the Chat widget appears on the pages of your website and the configuration buttons become available. Options: Yes / No|
```

File: src/configuration/dotdigital/developer.md [hunk 13]
```diff
@@ -96,3 +96,15 @@ _Cron Timings_
...
+## System Alerts
+
+![System Alerts]({% link images/images/dotdigital-dash-systemalerts.png %}){: .zoom}
+_System Alerts_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enable System Messages|Global|Determines if in-app system messages are sent to admin users when errors that may affect the dotdigital extension are detected. Options: Yes / No|
+|Enable Email Notifications|Global|Determines if system alert email notifications are sent to users with relevant permissions when errors that may affect the dotdigital extension are detected. Options: Yes / No|
+|Select Roles|Global|Specifies the user role(s) that will receive system alert email notifications. To select multiple roles, select the **Ctrl** key.|
```
Comment (ccondra) [inline src/configuration/dotdigital/developer.md:109]:
```suggestion
|Select Roles|Global|Specifies the user role(s) that receive system alert email notifications. To select multiple roles, select the **Ctrl** key.|
```

File: src/configuration/dotdigital/developer.md [hunk 14]
```diff
@@ -96,3 +96,15 @@ _Cron Timings_
...
+|Alert Frequency|Website|Specifies both the email notification frequency and the time period for when the extension reports errors. Options: 1 Hour / 12 Hours / 24 Hours / 48 Hours / 72 Hours|
```
Comment (ccondra) [inline src/configuration/dotdigital/developer.md:110]:
```suggestion
|Alert Frequency|Website|Specifies both the email notification frequency and the time interval between error reports. Options: 1 Hour / 12 Hours / 24 Hours / 48 Hours / 72 Hours|
```

**Review by ccondra — CHANGES_REQUESTED** [review]
Just a few minor edits.  That's it!  -- Thanks!

File: src/configuration/dotdigital/developer.md (hunk 14 above)
Comment (sta1r) [reply src/configuration/dotdigital/developer.md:110]: @ccondra This one is a bit tricky. Alert Frequency 24hrs means:
- you get emails once every 24 hours
- only errors from the last 24 hours will be flagged

We wrestled with the phrasing on this one as well.

Comment (ccondra) [reply src/configuration/dotdigital/developer.md:110]: Yes, it is tricky!  I added your example to make it clear. Thanks!

**Review by ccondra — APPROVED** [review]

**Comment by leslietilling** [comment]
running tests

**Pull request merged**